### DISCUSSION TITLE
mtproto client side implementation in vlang

### DISCUSSION THREAD

**Discussion by andrew-ld** [vlang/v#15188] [General]
hello, it would be extremely useful a mtproto client written entirely in vlang, so we combine the power of telegram and the compilation speed of vlang

specifications: https://core.telegram.org/mtproto
implementation in other languages: https://github.com/pyrogram/pyrogram, https://github.com/danog/MadelineProto

**Comment by Gladear** [comment]
This is very unlikely that this protocol will make it to the standard vlib.
You can wait for someone to create this and upload it to vpm, or implement it yourself if you feel like it 🙂

**Comment by MarcoBuster** [comment]
@andrew-ld YAAS!! Such an awesome idea!!!!
LET'S GOOOOOO vtdlight

**Comment by skrtdev** [comment]
@andrew-ld it does not make sense to have it in the vlib. I'm actually making a bot api library and I'll try to make tdlib bindings. I can also help if someone wants make a full mtproto client. However at the moment I find a new compiler bug every time I try to compile, so V it's not ready yet to have a mtproto client written in it

**Comment by nedpals** [comment]
@skrtdev was right. Modules are mostly like to be qualified to be included in vlib when they are frequently used by the users like `json` or generic libraries like `net`. There are exceptions (like `pico.v` and `picohttpparser`) but they will be reimplemented as what I've said, standalone generic libraries where people can build upon. 

In the case of mtproto, it does not satisfy the first and second requirement. Telegram is the only one I've known that used this library/protocol and the code for building bots cannot be also used in other platforms (unless we have marketed this language as "the best way write Telegram bots").

This issue won't be closed so that other users can see it. There are other modules request as well that you can see [here](https://github.com/vlang/v/labels/Module%20Request).
